src/Datadog.Trace.ClrProfiler.Native/cor_profiler.cpp

```diff
@@ -797,8 +793,14 @@ HRESULT CorProfiler::GenerateVoidILStartupMethod(const ModuleID module_id,
     return hr;
   }
 
+#ifdef _WIN32
+  WSTRING native_profiler_file = "DATADOG.TRACE.CLRPROFILER.NATIVE.DLL"_W;
+#else
+  WSTRING native_profiler_file = GetEnvironmentValue("CORECLR_PROFILER_PATH"_W);
```

Comment:
Is there any additional protection we should do here? In order for the profiler to be loaded, this must have been defined.

Reply:
Technically, users can define `CORECLR_PROFILER_PATH_32` or `CORECLR_PROFILER_PATH_64` instead of `CORECLR_PROFILER_PATH`. We might want define fallbacks depending on platform.

Reply:
Would that be for .NET Core in general?
And would we want to prioritize using the variable with the specific architecture over the general case?
e.g., 
If process is 32 bit, try getting `CORECLR_PROFILER_PATH_32` 
If process is 64 bit try getting `CORECLR_PROFILER_PATH_64`
If profiler path hasn't been found, fallback to `CORECLR_PROFILER_PATH`

Reply:
Yes, it's a corefx feature. netfx [supposedly supports](https://github.com/dotnet/coreclr/blob/master/Documentation/project-docs/clr-configuration-knobs.md) it, too (`COR_PROFILER_PATH_32`/`COR_PROFILER_PATH_64`), but the documentation does not specify in what version it was added (if it was even added at all).

And yes, we should try the platform-specific ones first and then fallback. [This is what the CLR does](https://github.com/dotnet/coreclr/blob/master/Documentation/Profiling/Profiler%20Loading.md#environment-variable-cross-plat), so if both are defined with different values, we have a better chance of using the correct one.

Reply:
Sounds good, I'll implement the bitness-preferred check with the general fallback.

As to why I'm doing it this way, it turns out that when CoreCLR does the library load code, it only finds images on the LD_LIBRARY_PATH, even though our native profiler is already executing. That means we either 1) load the profiler by passing in the exact file path or 2) we must add the profiler's directory to the user's LD_LIBRARY_PATH environment variable. I opted for the first choice.

Reply:
Fixed in the latest commit. Also, are the debug statements overkill?

Reply:
I verified on a local x64 Alpine container that the mechanism for selecting the library load path where we prefer `CORECLR_PROFILER_PATH_32|64` over `CORECLR_PROFILER_PATH` works 👍